test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -219,9 +222,14 @@ public IListen listen()
     public SimpleQueryResult executeInternalWithResult(String query, Object... args)
     {
         return sync(() -> {
+            ClientWarn.instance.captureWarnings();
+            CoordinatorTrackWarnings.init();
             QueryHandler.Prepared prepared = QueryProcessor.prepareInternal(query);
             ResultMessage result = prepared.statement.executeLocally(QueryProcessor.internalQueryState(),
                                                                      QueryProcessor.makeInternalOptions(prepared.statement, args));
+            CoordinatorTrackWarnings.done();
```

Comment:
Is it a problem if this doesn't fire if we get an exception from `executeLocally()`?